src/unix/index_unix.ml

```diff
@@ -151,9 +153,12 @@ module IO : Index.IO = struct
 
   let () = assert (String.length current_version = 8)
 
-  let v ~readonly ~fresh ~generation ~fan_size file =
+  let v ?auto_flush_callback ~readonly ~fresh ~generation ~fan_size file =
     let v ~fan_size ~offset raw =
       let header = 8L ++ 8L ++ 8L ++ 8L ++ fan_size in
+      let auto_flush_callback =
+        match auto_flush_callback with None -> Fun.id | Some f -> f
+      in
```

Comment:
```suggestion
  let v ?(auto_flush_callback = fun () -> ()) ~readonly ~fresh ~generation ~fan_size file =
    let v ~fan_size ~offset raw =
      let header = 8L ++ 8L ++ 8L ++ 8L ++ fan_size in
```

Not actually sure if it's better, but this is what I was imagining. As you prefer.